ingestion/functions/parsing/japan/japan.py

```diff
@@ -43,23 +43,48 @@ def convert_location(raw_entry):
         if term and term != "Unspecified"]
     return {"query":  ", ".join(query_terms)}
 
-
 def detect_notes(raw_notes):
     if "notes" in raw_notes:
         return raw_notes["notes"]
     else:
         return None
 
-
 def convert_date(raw_date: str):
     """
     Convert raw date field into a value interpretable by the dataserver.
-
+​
     The date filtering API expects mm/dd/YYYYZ format.
     """
-    date = datetime.strptime(raw_date, "%Y-%m-%d")
+    date = datetime.strptime(raw_date["dateAnnounced"], "%Y-%m-%d")
     return date.strftime("%m/%d/%YZ")
 
+def convert_death_date(raw_death_date: str):
+    if "deceasedDate" in raw_death_date:
+        death_date = datetime.strptime(raw_death_date["deceasedDate"], "%Y-%m-%d")
+        return death_date.strftime("%m/%d/%YZ")
+    else:
+        return None
+
+def convert_additionalSources(additional_sourceURL: str):
+    additionalSources_list = []
+    additionalSources = {}
+    if "sourceURL" in additional_sourceURL:
+        additionalSources["sourceUrl"] = additional_sourceURL["sourceURL"]
+    if "deathSourceURL" in additional_sourceURL:
```

Comment:
If you have both source and death urls the death one will overwrite the source one.
Consider rewriting this method so it's simpler:

```
sources = []
if "sourceURL" in additional_sourceURL:
  sources.append({"sourceUrl": additional_sourceURL["sourceURL"])
if "deathSourceURL" in additional_sourceURL:
  sources.append({"sourceUrl": additional_sourceURL["deathSourceURL"])
return sources or None
```